### PR TITLE
1604 by Inlead: Implement ding_sections to DDB CMS.

### PR DESCRIPTION
**Pull request by inlead**
http://platform.dandigbib.org/issues/1604

### CODE REVIEW

**Comment by guddo** [comment]
This PR is made for the 7.x-2.5.1 release.

**Comment by guddo** [comment]
This PR is made for the 7.x-2.5.1 release.

File: ding2.info [hunk 1]
```diff
@@ -1,5 +1,6 @@
 name = Ding 2
 description = Install a Ding based library site.
+version = 7.x-2.5.1
```
Comment (cableman) [inline ding2.info]: This version line should not be added to the PR.

**Comment by cableman** [comment]
Please write a few lines in the README file about what this module is used for and if it needs any configuration to run.

File: modules/ding_sections/ding_sections.admin.inc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+/**
+ * @file
+ * Ding section dashboard administrative interface.
+ */
+
+/**
+ * Creates an administrative block on the configuration page.
+ *
+ * It is necessary for grouping various settings.
+
```
Comment (cableman) [inline modules/ding_sections/ding_sections.admin.inc]: Missing * in comment linje

File: modules/ding_sections/ding_sections.admin.inc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  foreach ($node_types as $type) {
+    $node_bundles[$type->type] = $type->name;
+  }
+
+  $form['section_content_types'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('Available node types'),
+    '#description' => t('Select which nodes should be a part of section system'),
+    '#options' => $node_bundles,
+    '#default_value' => variable_get('section_content_types', array()),
```
Comment (cableman) [inline modules/ding_sections/ding_sections.admin.inc]: The variable (and index in $form) should be prefixed with ding_ (so it's ding_section_content_types).

File: modules/ding_sections/ding_sections.admin.inc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+/**
+ * Custom submit handler for content types admin form.
+ *
+ * Create hidden fields for selected content types that will contain
+ * section status of the node.
+ */
+function ding_sections_admin_content_types_submit($form, $form_state) {
+  $input = $form_state['input'];
+  $node_types = array_filter($input['section_content_types']);
+  $last_setting = array_filter(variable_get('section_content_types', array()));
```
Comment (cableman) [inline modules/ding_sections/ding_sections.admin.inc]: Remember to prefix the variable with "ding_" here as well :-)

File: modules/ding_sections/ding_sections.admin.inc [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+  if (count($node_types) == 0 && empty($last_setting)) {
```
Comment (cableman) [inline modules/ding_sections/ding_sections.admin.inc]: Could this count use empty() instead as with the $last_setting variable?

File: modules/ding_sections/ding_sections.admin.inc [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    return;
+  }
+
+  $instance_definition = array(
+    'label' => 'Section',
+    'field_name' => 'field_section',
```
Comment (cableman) [inline modules/ding_sections/ding_sections.admin.inc]: Should this have been prefixed with "ding_" or "field_ding_"

File: modules/ding_sections/ding_sections.info [hunk 7]
```diff
@@ -0,0 +1,5 @@
+name = Ding Sections
+description = Provides integration of content in the section system.
```
Comment (cableman) [inline modules/ding_sections/ding_sections.info]: It might be me... but what is "the selection system"?

File: modules/ding_sections/ding_sections.install [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+/**
+ * Implements hook_install().
+ */
+function ding_sections_install() {
+  _ding_sections_create_base();
+  _ding_sections_create_fields();
+  ding_sections_section_shortcut();
+
+  variable_set('section_content_types', array());
```
Comment (cableman) [inline modules/ding_sections/ding_sections.install]: Please prefix with "ding_".

File: modules/ding_sections/ding_sections.module [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+ *   Loaded vocabulary.
+ *
+ * @return bool
+ *   Access value.
+ */
+function ding_sections_section_add_access($voc) {
+  global $user;
+
+  if ($voc->machine_name == 'section') {
+    $access = user_access('edit terms in 8', $user);
```
Comment (cableman) [inline modules/ding_sections/ding_sections.module]: Will this work is it always "8"?

File: modules/ding_sections/ding_sections.module [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+    'title' => 'Use the Panels In-Place Editor for terms',
+  );
+
+  return $perm;
+}
+
+/**
+ * Implements hook_secure_permissions().
+ */
+function ding_sections_secure_permissions($role) {
```
Comment (cableman) [inline modules/ding_sections/ding_sections.module]: Please move permissions to ding_permissions module.

File: modules/ding_sections/modules/ding_sections_custom_css/ding_sections_custom_css.features.field_base.inc [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+/**
+ * Implements hook_field_default_field_bases().
+ */
+function ding_sections_custom_css_field_default_field_bases() {
+  $field_bases = array();
+
+  // Exported field_base: 'field_custom_css'
+  $field_bases['field_custom_css'] = array(
```
Comment (cableman) [inline modules/ding_sections/modules/ding_sections_custom_css/ding_sections_custom_css.features.field_base.inc]: Field should be prefixed with module name "field_ding_sections_".

File: modules/ding_sections/modules/ding_sections_header_image/ding_sections_header_image.features.field_base.inc [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+/**
+ * Implements hook_field_default_field_bases().
+ */
+function ding_sections_header_image_field_default_field_bases() {
+  $field_bases = array();
+
+  // Exported field_base: 'field_header_image'
+  $field_bases['field_header_image'] = array(
```
Comment (cableman) [inline modules/ding_sections/modules/ding_sections_header_image/ding_sections_header_image.features.field_base.inc]: Fields should be prefixed with module name.

File: modules/ding_sections/modules/ding_sections_term_menu/ding_sections_term_menu.install [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * @file
+ * Code for module installation.
+ */
+
+/**
+ * Implements hook_schema().
+ */
+function ding_sections_term_menu_schema() {
+  $schema['section_term_menu'] = array(
```
Comment (cableman) [inline modules/ding_sections/modules/ding_sections_term_menu/ding_sections_term_menu.install]: Please prefix scheme with module name.

File: modules/ding_sections/modules/ding_sections_term_panel/ding_sections_term_panel.module [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+  $task = page_manager_get_task('term_view');
+  $handlers = page_manager_load_task_handlers($task, '');
+
+  if (!isset($handlers[$handler_name])) {
+    return FALSE;
+  }
+
+  $handler = page_manager_export_task_handler($handlers[$handler_name]);
+
+  ob_start();
```
Comment (cableman) [inline modules/ding_sections/modules/ding_sections_term_panel/ding_sections_term_panel.module]: This don't seam right to ever use eval?

File: themes/ddbasic/template.php [hunk 15]
```diff
@@ -168,7 +168,7 @@ function ddbasic_preprocess_panels_pane(&$vars) {
       // Menu-block.
       $vars['content']['#content']['#theme_wrappers'] = array('menu_tree__sub_menu');
     }
-    else {
+    elseif (isset($vars['content']['#theme_wrappers'])) {
```
Comment (cableman) [inline themes/ddbasic/template.php]: Not sure why this change is needed?

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

 * Commit: 142f783a977f80ac8496a44e2a8aba5bad93fe2a
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: e72bae53daf3bd0ba5f6fb0229e7e1a3f4a339ba
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: a433f36e20991a37e58c3f69c47a4e45b24da667
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by inlead** [comment]
Closed in favour of https://github.com/ding2/ding2/pull/1047.
